ci/spec.yml

```diff
@@ -134,19 +134,18 @@ jobs:
       - *copy_secret
 
       - name: Regenerate workflow YAML file and check for changes
-        run: |
-          set -e
-          pip install pyyaml
-          ./scripts/build_ensure_no_changes.sh ./ci/regen-workflow.py
+        run: ./scripts/build_ensure_no_changes.sh ./ci/regen-workflow.py
 
       # Build docs/sources
       - name: Build docs
         run: ./scripts/build_ensure_no_changes.sh ./scripts/build_docs.sh
       - name: Build sources
-        run: ./scripts/build_ensure_no_changes.sh ./scripts/build_sources.sh
+        run: |
```

Comment:
Doesn't the virtual env already get activated earlier?

Reply:
Apparently not?
Or at least when I hacked up build_sources.sh to pip show setuptools, it was the 47 build, not the 57 one installed in the venv.
I'm not sure *why* the venv isn't activated, but it seems that something is missing...

Reply:
Hmm that's really weird. I wouldn't think it would work at all when running CI if the venv wasn't active. So if you run CI without (re?)activating the venv you see a different version of setuptools than if you don't?

Reply:
yup.
Although now that I look closer, the shebang for manage.py is python3 and setup.py is python, so it might be that the venv only overwrites the former and the latter pull in system python2. Let me try changing that and see if that solves this...

Reply:
Nope, that wasn't it.
Given that it doesn't seem like venvs are used at all in the official github example, I'm wondering if it's even worth having one here...
ref https://docs.github.com/en/actions/guides/building-and-testing-python

Reply:
Yea, if the build section works without a venv, I don't see anything on casual inspection that would necessitate it.